src/ring_theory/power_series/basic.lean

```diff
@@ -450,42 +450,25 @@ variables [comm_semiring R] (n : σ →₀ ℕ)
 
 /-- Auxiliary definition for the truncation function. -/
 def trunc_fun (φ : mv_power_series σ R) : mv_polynomial σ R :=
-{ support := (n.antidiagonal.support.image prod.fst).filter (λ m, coeff R m φ ≠ 0),
-  to_fun := λ m, if m ≤ n then coeff R m φ else 0,
-  mem_support_to_fun := λ m,
-  begin
-    suffices : m ∈ finset.image prod.fst ((antidiagonal n).support) ↔ m ≤ n,
-    { rw [finset.mem_filter, this], split,
-      { intro h, rw [if_pos h.1], exact h.2 },
-      { intro h, split_ifs at h with H H,
-        { exact ⟨H, h⟩ },
-        { exfalso, exact h rfl } } },
-    rw finset.mem_image, split,
-    { rintros ⟨⟨i,j⟩, h, rfl⟩ s,
-      rw finsupp.mem_antidiagonal_support at h,
-      rw ← h, exact nat.le_add_right _ _ },
-    { intro h, refine ⟨(m, n-m), _, rfl⟩,
-      rw finsupp.mem_antidiagonal_support, ext s, exact nat.add_sub_of_le (h s) }
-  end }
+∑ m in Iic_finset n, mv_polynomial.monomial m (coeff R m φ)
```

Comment:
If `monomial` and `coeffs` were `add_monoid_hom`s, then you could make this even shorter as
```suggestion
def trunc_fun : mv_power_series σ R →+ mv_polynomial σ R
∑ m in Iic_finset n, (mv_polynomial.monomial m).comp (coeff R m)
```

Reply:
But this PR is great already!

Reply:
The downside of bundling is that you can't use dot notation. It makes sense (to me) to bundle `monomial`, but writing `f.coeff n` can be nice...
Still, I think a PR that bundles coeff is a good idea.

Reply:
Even a separate bundled definition would be enough to use in this definition.

Bundling `monomial` is probably awkward too, since we have simp lemmas on monomial which act in the reverse direction to `monomial.map_add`...